stdlib/pair.mli

```diff
@@ -0,0 +1,77 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*                         The OCaml programmers                          *)
+(*                                                                        *)
+(*   Copyright 2018 Institut National de Recherche en Informatique et     *)
+(*     en Automatique.                                                    *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+(** Function on pairs of values. *)
+
+
+(** {1 Pairs} *)
+
+type ('a, 'b) t = 'a * 'b
+(** The type for pairs. *)
+
+val make: 'a -> 'b -> ('a, 'b) t
+(** [make a b] is the pair [(a, b)]. *)
+
+val swap: ('a, 'b) t -> ('b, 'a) t
+(** [swap (a, b)] is [(b, a)]. *)
+
+(** {1 Iterators} *)
+
+val fold: ('a -> 'b -> 'c) -> ('a, 'b) t -> 'c
+(** [fold f (a, b)] applies [f] to [a] and [b]. *)
+
+val uncurry: ('a -> 'b -> 'c) -> ('a, 'b) t -> 'c
+(** [uncurry] is the same as {!fold}. *)
```

Comment:
`Pair.uncurry` reads strangely to me. I would propose to drop the function since it is a duplicate for `fold`.

Reply:
`curry` and `uncurry` are standard haskellisms.

I agree it should be dropped: if desirable such functions would likely make more sense in the `Fun` module.